spec/model/user_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is not valid without a name' do
+    first_user = User.new(name: '', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    expect(first_user).not_to be_valid
+  end
+
+  it 'is valid without a name' do
+    first_user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    expect(first_user).to be_valid
+  end
+
+  it 'gets the first 3 posts' do
+    first_user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    Post.create(author: first_user, title: 'post 1', text: 'This is my first post')
+    Post.create(author: first_user, title: 'post 2', text: 'This is my second post')
+    third_post = Post.create(author: first_user, title: 'post 3', text: 'This is my third post')
+    fourth_post = Post.create(author: first_user, title: 'post 4', text: 'This is my fourth post')
+    fifth_post = Post.create(author: first_user, title: 'post 5', text: 'This is my fifth post')
+    result = first_user.top_posts
+    puts result.class
+    expect(first_user.top_posts).to eq([fifth_post, fourth_post, third_post])
+  end
+
+  it 'gets the Users post_count' do
+    first_user = User.new(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')
+    Post.create(author: first_user, title: 'post 1', text: 'This is my first post')
+    Post.create(author: first_user, title: 'post 2', text: 'This is my second post')
+    Post.create(author: first_user, title: 'post 3', text: 'This is my third post')
+    Post.create(author: first_user, title: 'post 4', text: 'This is my fourth post')
+    Post.create(author: first_user, title: 'post 5', text: 'This is my fifth post')
+    expect(first_user.post_counter).to eq(5)
+  end
+end
```

Comment:
- [ ]  Hi, well done for writing the test. I would advise you to add a test for `comment` and `like` because the requirements say: **Add unit specs for all of your models' methods and validations.**

Reply:
@OLIPLICHE  Thanks soo much I have implemented the changes now, by adding test for comment and like